packages/cli/src/inspectWriteGenerate.ts

```diff
@@ -105,16 +107,24 @@ export async function inspectWriteGenerate({
   } else {
     defaultConfig.introspection.endpoint = DUMMY_ENDPOINT;
 
-    if (existsSync(endpoint)) {
-      const file = await promises.readFile(endpoint, {
-        encoding: 'utf-8',
-      });
+    const files = await fg(endpoint);
+    if (files.length) {
+      const fileContents = await Promise.all(
+        files.map((file) =>
+          promises.readFile(file, {
+            encoding: 'utf-8',
+          })
+        )
+      );
 
       if (endpoint.endsWith('.json')) {
         const parsedFile:
           | IntrospectionQuery
           | { data?: IntrospectionQuery }
-          | undefined = JSON.parse(file);
+          | undefined = fileContents.reduce(
+          (acc, file) => merge(acc, JSON.parse(file)),
```

Comment:
```suggestion
          (acc, file) => ({
            ...acc,
            ...JSON.parse(file),
          }),
```

I see you are still working on it, but honestly lodash, or lodash.anything, does not worth the bundle bloat.

Do you think it works for the suggested change without lodash?

Reply:
Fine by me - made the change.
I don't personally know how this affects merging introspection JSONs but I also don't think anyone will use it with multiple JSONs - its more a use case of using multple .gql schema files anyways so I believe this should be fine